north/src/linux/inotify.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019 - 2020 ESRLabs
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+use anyhow::{anyhow, Result};
+use async_std::task;
+use nix::{
+    self,
+    errno::Errno,
+    sys::inotify::{AddWatchFlags, InitFlags, Inotify},
+};
+use std::fs::metadata;
+use std::path::Path;
+use std::time::Duration;
+
+pub async fn wait_for_file_deleted(path: &Path, timeout: Duration) -> Result<()> {
+    let my_path = path.to_owned();
+    task::spawn_blocking(move || {
+        let inotify = Inotify::init(InitFlags::IN_CLOEXEC | InitFlags::IN_NONBLOCK)?;
+        inotify.add_watch(&my_path, AddWatchFlags::IN_DELETE_SELF)?;
+
+        let start_time = std::time::SystemTime::now();
+
+        loop {
```

Comment:
Is this spinning until the file is removed? 
`inotify` offers a blocking read that avoids this.